R/fit-models.R

```diff
@@ -99,48 +94,17 @@ build_formula <- function(target, covariates, smooth_terms = NULL, group_by = "g
 #'                    k = "auto")
 #' }
 fit_gam <- function(df_tract,
-                    target = NULL,
+                    target,
                     covariates = NULL,
-                    smooth_terms = NULL,
-                    group_by = NULL,
-                    participant_id = NULL,
+                    additional_terms = NULL,
+                    group_by = "group",
+                    participant_id = "subjectID",
                     formula = NULL,
-                    k = NULL,
+                    k = 40,
                     family = "auto",
-                    method="fREML",
-                    ...) {
-
-  # Check that if formula is non-NULL, all the other formula-setting inputs
```

Comment:
Why is this code being removed? This code is here to deal with the two options: provide a formula, or let the code build the formula from the inputs. Unless I am missing something, I don't think that the current PR should be eliminating these options.